src/os/bluestore/BlueStore.cc

```diff
@@ -8812,6 +8812,7 @@ int BlueStore::_do_alloc_write(
     // counting machinery.
     assert(!b->is_referenced());
     b->get_ref(coll.get(), wi.b_off0, wi.length0); 
+    b->_encode_blob_to_cache(3);
```

Comment:
Thanks for pointing this out. I planned to make a broader cleanup of the `struct_v` usage but haven't made it yet. :-(

I think the `struct_v` in the `BlueStore::Blob::bound_encode` and `BlueStore::Blob::encode` paths is unnecessary. It's actually used only by asserts in `bluestore_blob_t`. The version seems to be [encoded](https://github.com/ceph/ceph/blob/master/src/os/bluestore/BlueStore.cc#L2374) solely in `BlueStore::ExtentMap::encode_some`.

In the decode path `BlueStore::ExtentMap::decode_some` [extracts](https://github.com/ceph/ceph/blob/master/src/os/bluestore/BlueStore.cc#L2445) it and [passes](https://github.com/ceph/ceph/blob/master/src/os/bluestore/BlueStore.cc#L2490) to `BlueStore::Blob::decode` that is a single place [dependant](https://github.com/ceph/ceph/blob/master/src/os/bluestore/BlueStore.cc#L1908-L1920) on that.